frontend/src/components/Forms/RegisterForm/RegisterForm.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from "react";
+import { SubmitHandler, useForm, FormProvider } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import { useNavigate } from "react-router-dom";
+
+import TextFieldInput from "../../Fields/FormInput/TextFieldInput";
+import fetchPlus from "../../../helpers/fetchPlus";
+import Constants from "../../../helpers/Constants";
+import StatusCode from "../../../helpers/StatusCode";
+
+const validationSchema = z
+  .object({
+    email: z
+      .string()
+      .min(1, { message: "Email is required" })
+      .email({ message: "Please enter a valid email address" }),
+    password: z
+      .string()
+      .min(1, "Password is required")
+      .min(8, "Password must be more than 8 characters")
+      .max(32, "Password must be less than 32 characters"),
+    confirmPassword: z
+      .string()
+      .min(1, { message: "Confirm Password is required" }),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    path: ["confirmPassword"],
+    message: "The passwords don't match",
+  });
+
+type ValidationSchema = z.infer<typeof validationSchema>;
+
+function RegisterForm() {
+  const navigate = useNavigate();
+  const defaultValues: ValidationSchema = {
+    email: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  const methods = useForm<ValidationSchema>({
+    resolver: zodResolver(validationSchema),
+    defaultValues,
+  });
+
+  const onSubmitHandler: SubmitHandler<ValidationSchema> = async (
+    data: ValidationSchema
+  ) => {
+    fetchPlus(`${Constants.BASE_URL}/auth/register`, {
+      body: JSON.stringify({
+        email: data.email,
+        password: data.password,
+      }),
+    })
+      .then((response: Response) => {
```

Comment:
Why .then and not just await since this is already marked as an async method?

Reply:
I forgot to delete the `async` mark after some testing. Fixed that.

Reply:
As we discussed, I have changed the `.then` to `await`.